test/index.test.js

```diff
@@ -19,48 +57,127 @@ describe('desktop-trampoline', () => {
   it('fails when required environment variables are missing', () =>
     expect(run(trampolinePath, ['Username'])).rejects.toThrow())
 
-  it('forwards arguments and valid environment variables correctly', async () => {
-    const output = []
-    const server = createServer(socket => {
-      socket.pipe(split2(/\0/)).on('data', data => {
-        output.push(data.toString('utf8'))
-      })
+  describe('with a trampoline server', () => {
+    let server = null
+    let output = []
+    let baseEnv = {}
 
-      // Don't send anything and just close the socket after the trampoline is
-      // done forwarding data.
-      socket.end()
+    async function configureTrampolineServer(stdout = '', stderr = '') {
```

Comment:
Unfortunately this cannot be a `beforeEach` function (or I didn't see how), since we need to pass parameters when the server is created 😞